graphql-dgs/src/main/kotlin/com/netflix/graphql/dgs/context/DgsContext.kt

```diff
@@ -24,7 +25,7 @@ import org.dataloader.BatchLoaderEnvironment
  * Context class that is created per request, and is added to both DataFetchingEnvironment and BatchLoaderEnvironment.
  * Custom data can be added by providing a [DgsCustomContextBuilder].
  */
-open class DgsContext(val customContext: Any? = null, val requestData: DgsRequestData?) {
+open class DgsContext(val customContext: Any? = null, val requestData: DgsRequestData?) : HTTPRequestHeaders {
```

Comment:
The missing link for me why we would want to mark the context as such is [here](https://github.com/apollographql/federation-jvm/blob/62edcdddecc01ea4225cc88efc18c1b4197a911a/graphql-java-support-api/src/main/java/com/apollographql/federation/graphqljava/tracing/HTTPRequestHeaders.java#L6). I would elaborate on this w/ a self-comment in PR or in the code to help reviewers understand why.